pkg/validate/raw/validate/source_spec_validator.go

```diff
@@ -46,48 +46,48 @@ func HelmValuesFileDataKeyOrDefault(key string) string {
 }
 
 // RepoSyncSpec validates the Repo Sync source specification for any obvious problems.
-func RepoSyncSpec(sourceType configsync.SourceType, git *v1beta1.Git, oci *v1beta1.Oci, helm *v1beta1.HelmRepoSync, rs client.Object) status.Error {
+func RepoSyncSpec(sourceType configsync.SourceType, git *v1beta1.Git, oci *v1beta1.Oci, helm *v1beta1.HelmRepoSync, syncKind string) status.Error {
```

Comment:
Is this used elsewhere, besides repo_sync_validator.go?

Reply:
Yeah, in the reconciler-manager's reposync controller. 

The reconciler-manager validates the RSyncs specs. That's why we don't need the reconciler to do it when it finds an RSync in a source repo.

Reply:
Where is `RepoSyncSpec` called in `reposync_controller.go`? I didn't see it.

Reply:
Good catch. Looks like `reposync_controller.go`  does call all the underlying functions, but not that parent function.

```

func (r *RepoSyncReconciler) validateSourceSpec(ctx context.Context, rs *v1beta1.RepoSync, reconcilerName string) error {
	switch rs.Spec.SourceType {
	case configsync.GitSource:
		return r.validateGitSpec(ctx, rs, reconcilerName)
	case configsync.OciSource:
		return r.validateOciSpec(ctx, rs)
	case configsync.HelmSource:
		return r.validateHelmSpec(ctx, rs)
	default:
		return validate.InvalidSourceType(r.syncGVK.Kind)
	}
}
```

Basically the same thing, but with a few extra checks.

RepoSyncSpec & RootSyncSpec have good test coverage tho. Not sure how best to break it up.

Reply:
How about adding a comment saying that this function is no longer used for validation, and is only kept for its test coverage for the underlying validations. 

This change LGTM otherwise.

Reply:
I did some refactoring to make the controllers use the tested validate methods instead. It also reduced some redundancy. And I added a basic non-negative check for resource quantities. It's not clear why they allow negatives upstream, but AFAICT they fail to reconcile when the container runtime tries to create the container.